optional.h

```diff
@@ -23,17 +40,21 @@ class Optional {
 	}
 
 	T& get() {
-		assert(m_valid);
+		if (!m_valid) {
+			throw BadOptional();
+		}
 		return *m_value;
 	}
 		
 	const T& get() const {
-		assert(m_valid);
+		if (!m_valid) {
+			throw BadOptional();
+		}
 		return *m_value;
 	}
 	
 private:
-	char buf[sizeof(T)];
+	alignas(alignof(T)) char buf[sizeof(T)];
```

Comment:
Great! This is the right solution.
(Just note that we have `std::aligned_storage` exactly for this purpose.)